astropy/visualization/units.py

```diff
@@ -51,28 +52,32 @@ def rad_fn(x, pos=None):
             return f"{n}π/2"
 
     class MplQuantityConverter(units.ConversionInterface, ContextDecorator):
+        axis_labels_val = outer_args
         def __init__(self):
             # Keep track of original converter in case the context manager is
             # used in a nested way.
             self._original_converter = {u.Quantity: units.registry.get(u.Quantity)}
             units.registry[u.Quantity] = self
+                        
 
         @staticmethod
         def axisinfo(unit, axis):
+            
```

Comment:
Perhaps `axisinfo` should become a regular method, so we have access to `self` here?